trieste/models/keras/models.py

```diff
@@ -0,0 +1,293 @@
+    as later work that more precisely measured uncertainty quantification found that boostrapping
+    does help with uncertainty representation (see <cite data-cite="osband2021epistemic"/>).
+
+    We provide classes for constructing ensembles using Keras
+    (:class:`~trieste.models.keras.KerasEnsemble`) in the `architectures` package that should be
+    used with the :class:`~trieste.models.keras.DeepEnsemble` wrapper. There we also provide a
+    :class:`~trieste.models.keras.GaussianNetwork` base learner following the original
+    formulation in <cite data-cite="lakshminarayanan2017simple"/>, but any user-specified network
+    can be supplied, as long as it has a Gaussian distribution as a final layer and follows the
+    :class:`~trieste.models.keras.KerasEnsembleNetwork` interface.
+
+    Note that setting up the model with dictionary configs is not currently possible.
+
+    When using the model we recommend to set `tf.keras.backend.set_floatx(tf.float64)` for
+    alignment with the Trieste toolbox.
+    """
+
+    def __init__(
+        self,
+        model: KerasEnsemble,
+        optimizer: Optional[KerasOptimizer] = None,
+        bootstrap: bool = False,
+    ) -> None:
+        """
+        :param model: A Keras ensemble model with probabilistic networks as ensemble members. The
+            model has to be built but not compiled.
+        :param optimizer: The optimizer wrapper with necessary specifications for compiling and
+            training the model. Defaults to :class:`~trieste.models.optimizer.KerasOptimizer` with
+            :class:`~tf.optimizers.Adam` optimizer, negative log likelihood loss and a dictionary
+            of default arguments for Keras `fit` method: 100 epochs, batch size 100, and verbose 0.
+            See https://keras.io/api/models/model_training_apis/#fit-method for a list of possible
+            arguments.
+        :param bootstrap: Sample with replacement data for training each network in the ensemble.
+            By default set to `False`.
+        :raise ValueError: If ``model`` is not an instance of
+            :class:`~trieste.models.keras.KerasEnsemble` or ensemble has less than two base
+            learners (networks).
+        """
+
+        if not isinstance(model, KerasEnsemble):
+            raise ValueError(
+                f"In DeepEnsemble model must be an instance of a KerasEnsemble, "
+                f"received {type(model)} instead."
+            )
+
+        if optimizer is None:
+            optimizer = KerasOptimizer(tf.keras.optimizers.Adam())
+        self._optimizer = optimizer
+
+        if not isinstance(optimizer.optimizer, tf.keras.optimizers.Optimizer):
+            raise ValueError(
+                f"Optimizer for `DeepEnsemble` model must be an instance of a "
+                f"`tf.keras.optimizers.Optimizer`, "
+                f"received {type(optimizer.optimizer)} instead."
+            )
+
+        super().__init__(optimizer)
```

Comment:
(if we default to using `KerasOptimizer` in `NeuralNetworkPredictor` then you can remove all the lines before this)